File: pharmacy-frontend/src/app/models/order.ts
// src/app/models/order.ts
import { CartItem } from './cart-item';

export interface OrderItem {
  productId: number;
  quantity: number;
}

export interface Order {
  id: number;
  status: string;
  customerEmail: string;
  createdAt: string;
  items: OrderItem[];
  reservationId?: number;
}

File: EONIS/Models/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EONIS.Models
{
    public class Order
    {
        public int Id { get; set; }

        [MaxLength(32)]
        public string Status { get; set; } = "Pending"; 

        [EmailAddress]
        public string? CustomerEmail { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();

    }
}
